checkov/terraform/checks/resource/gcp/GKEPodSecurityPolicyEnabled.py

```diff
@@ -1,17 +1,41 @@
-from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
-from checkov.common.models.enums import CheckCategories
+from __future__ import annotations
 
+from typing import Any
+
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckCategories, CheckResult
+
+
+class GKEPodSecurityPolicyEnabled(BaseResourceCheck):
+
+    """
+     Pod Security Policy was removed from GKE clusters with version >= 1.25.0
+    """
 
-class GKEPodSecurityPolicyEnabled(BaseResourceValueCheck):
     def __init__(self):
         name = "Ensure PodSecurityPolicy controller is enabled on the Kubernetes Engine Clusters"
         id = "CKV_GCP_24"
         supported_resources = ['google_container_cluster']
         categories = [CheckCategories.KUBERNETES]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def get_inspected_key(self):
-        return "pod_security_policy_config/[0]/enabled"
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+
+        if conf.get('min_master_version') and isinstance(conf.get('min_master_version'), list):
```

Comment:
@JamesWoolfenden Shouldn't we take into account the `master_version` ?  `min_master_version` is optional

Reply:
the fix i think is just for the edge case where it could possible be 1.24 or below as this check is now mostly irrelevant